app/models/game.rb

```diff
@@ -1,3 +1,12 @@
 class Game < ActiveRecord::Base
   enum status: [:safe, :check, :checkmate]
+
+  def self.board_state
+    board = []
+    Piece.all.each do |piece|
+      board << [piece.x_coordinate, piece.y_coordinate]
+    end
+    return board
+  end
```

Comment:
I like the idea of this. Needs just a little change.

Currently this is a `Class` method. Rather than an `instance` method. You want it to be an `instance` method, so that it can be called on a specific instance of a game.

## Class vs Instance
To remember difference between a Class and an Instance. I think of it like this.
Imagine we have a class called `Human`. 

I am a `Human`, so I am an `instance` of `Human`

```Ruby
mike = Human.new(name: 'mike')
mike # is now an instance of the Human class
```